### PR TITLE
NetworkDb: remove stale channels in batch

### PR DESCRIPTION
**Pull request by sstone**
Use batch queries to remove/prune multiple channels  (we currently iterate and use one query per channel). This should allow us to remove stale channels on Android  much more efficiently.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/db/NetworkDb.scala [hunk 1]
```diff
@@ -49,7 +50,9 @@ trait NetworkDb {
 
   def listChannelUpdates(): Seq[ChannelUpdate]
 
-  def addToPruned(shortChannelId: ShortChannelId)
+  def addToPruned(shortChannelId: ShortChannelId): Unit = addToPruned(Seq(shortChannelId))
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/db/NetworkDb.scala]: Why did you keep `def addToPruned(shortChannelId: ShortChannelId)`?

Comment (sstone) [reply eclair-core/src/main/scala/fr/acinq/eclair/db/NetworkDb.scala]: No real reason, I've removed it.

File: eclair-core/src/test/scala/fr/acinq/eclair/db/SqliteNetworkDbSpec.scala [hunk 2]
```diff
@@ -104,19 +106,36 @@ class SqliteNetworkDbSpec extends FunSuite {
     db.updateChannelUpdate(channel_update_1)
   }
 
-  test("add/remove/test pruned channels") {
+  test("remove many channels") {
```
Comment (pm47) [inline eclair-core/src/test/scala/fr/acinq/eclair/db/SqliteNetworkDbSpec.scala]: Why is this test so much more complicated?

File: eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala [hunk 3]
```diff
@@ -344,7 +344,7 @@ class Router(nodeParams: NodeParams, watcher: ActorRef, initialized: Option[Prom
 
     case Event(TickPruneStaleChannels, d) =>
       // first we select channels that we will prune
-      val staleChannels = getStaleChannels(d.channels.values, d.updates)
+      val staleChannels = getStaleChannels(d.channels.values, d.updates).toSeq
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala]: can we make take an `removeChannels` take an `Iterable` to remove a copy?

**Pull request merged**